db/migrate/20190912215625_add_listing_id_to_ahoy_messages.rb

```diff
@@ -0,0 +1,5 @@
+class AddListingIdToAhoyMessages < ActiveRecord::Migration[5.2]
+  def change
+    add_column :ahoy_messages, :listing_id, :integer
```

Comment:
Seems like a `belongs_to/has_many` relation between `ClassifiedListing`s and `EmailMessage`s 
You can specify a condition on the relation like [`where(utm_campaign: "Listing")`](https://guides.rubyonrails.org/association_basics.html#scopes-for-has-many-where). You'll be able to do the `includes(:email_messages)` on `ClassifiedListing`s in the controller instead of executing a query inside every iteration of the loop.
We'll also need an index on the `listing_id` column.

Reply:
I added the relationship between `ClassifiedListing` and `EmailMessage` as well as specified `-> where { utm_campaign: "Listing" }` in the model to make the query in the internal Classified Listing controller unnecessary. 

Would `includes(:email_messages)` be included in referencing listings `@classified_listings = ClassifiedListing.includes(:email_messages).xyz(abc)...` ?